src/Exchange.ts

```diff
@@ -202,6 +207,53 @@ export class Exchange extends AbstractContract {
         return orders;
     }
 
+    public placeSellTokenOrder(price: number, amount: number): Transaction {
```

Comment:
as always, I suggest using integer bignums in lowest denomination (`BN`) consistently throughout the entire library, which would eliminate the need for the multiplication and integer checks below.

Reply:
1. We use `BigNumber` and not `BN` - which is already an issue. But  web3.js is keep changing what do they use and how. There is an [ongoing discussion](<https://github.com/ethereum/web3.js/issues/2171>) about. Shall we wait for that to settle before we refactor?

1. As always  I'm for dev UX to `augmint-js` consumers.
I.e. in case value can be represented with number then I would prefer 
   ```js
    exchange.placeSellTokenOrder(1.01, 100.12)
    ```
    over 
    ```js
    const decimalsDiv  = await exchange.token.decimalsDiv;
    exchange.placeSellTokenOrder(1.01, decimalsDiv.mul(100.12) )
    ```
Not a huge difference just seems easier.

Reply:
Linked in the proposal, ECMAScript `BigInt` seems to be best approach (being standardized in the language), with consideration given to more specialized sub-types (eg. wei). Since the point would be consistency, I'm happy to defer this to a later effort in a separate PR.

However, in the meantime let's at least get rid of the copy-pasted integrity checks and replace them with something that performs both unit conversions and integrity checks at the same time.

So instead of
```
const amountToSend: number = amount * DECIMALS_DIV;
if (Math.round(amountToSend) !== amountToSend) {
	            throw new InvalidTokenAmountError(
	                ` placeSellTokenOrder error: provided price of ${amountToSend} has more decimals than allowed by AugmintToken decimals of ${DECIMALS}`
	            );
	        }
```

I would rather see something along the lines of:
```
const bnAmountToSend = AugmintToken.toRawAmount(amount);
```

Reply:
sum of chat: we use `BN` class from `bn.js` (see latest commit)